### PR TITLE
feat: Texture world geometry

### PR DESCRIPTION
**Pull request by michaeleggers**
As the PR's name implies. world geometry is now textured.
Notice that VALVE 220 format is expected (`polysoup.cpp`). 
- The engine supports all of the texturing features available in TrenchBroom (scaling, offsetting, rotating). 
- Face flags (emissiveness, transparency, ...) are not supported at this point.
- TGA textures are required! This will be fixed by another PR I am working on right now that will also support `jgp` and `png`. 
- To calculate the UVs correctly, the texture dimensions are needed. There is a new class `CImage` that just loads an image file from disk, that's it.  At the moment this is a bit dumb because the image data gets read, just to check the textures' dimensions and then gets deleted. We could definitely cache all the pixeldata when doing this in `polysoup` and use them for the actual texture data later! It is an optimization that I want to defer to the point when the game is running. It is a 'low hanging fruit' optimization. Just something to keep in mind!
- The renderer uses the 'batch draw commands' that I thought were a premature optimization. But it turned out to be the right thing to do: All the world geometry gets loaded into a *single* GPU memory object (in OpenGL terms a VBO for vertex data). Then, a render command is generated for each texture-type at startup time. Then, each frame we bind the whole GPU buffer containing all the world-triangles, go through these render-commands that just store the offset and length into the GPU buffer and the texture gets bound and the triangles get drawn. This reduces the draw-calls to the amount of textures being used in the world rather than drawing each triangle individually.
- For now, **only** the static world geometry is being drawn. I have to think about how to deal with brush- and point-entities and talk to @benekoehler again how to deal with those.
- The assets are excluded in the `.gitignore` from now on, apart from the `shaders` dir. We should use the SFTP to get new textures. I uploaded the map `temple2.map` and the required textures to the SFTP.

### CODE REVIEW

File: game.cpp [hunk 1]
```diff
@@ -351,49 +355,55 @@ bool Game::RunFrame(double dt) {
...
+#if 0
         // Render World geometry
-        m_Renderer->ImDrawTriPlanes(m_World.m_TriPlanes.data(), 
-                                    m_World.m_TriPlanes.size(), 
-                                    true,
-                                    DRAW_MODE_SOLID);
+        renderer->ImDrawMapTris(m_World.m_Tris.data(), 
+                                  m_World.m_Tris.size(), 
+                                  true,
+                                  DRAW_MODE_SOLID);
```
Comment (benekoehler) [inline game.cpp:385]: noob question: why do we need to register the tris with the renderer and afterwards pass them again into the draw call? couldn't we just save the tris in the renderer for static data?

Comment (michaeleggers) [reply game.cpp:385]: Not a noob question. It was old code (that is not compiled due to `#if 0`). Thanks for spotting this! Removed.

**Review by NicoB-Coding — COMMENTED** [review]
image
Runs on AMD GPU Config (5700 XT Red Devil)

**Comment by michaeleggers** [comment]
> Runs on AMD GPU Config (5700 XT Red Devil)

OK. Merging this now. Appears to be ok on Linux/AMD-GPU, Windows AMD/Nvidia GPUs.

**Pull request merged**